2.13/tests/test/test.js

```diff
@@ -497,57 +489,69 @@ function testAuthentication(handler, verb) {
                     .expect(401, done);
             }
         })
-    }    
+    }
 }
 
 function testAPIVersionHeader(handler, verb) {
     it('should reject requests without X-Broker-API-Version header with 412', function(done) {
         if (verb == 'GET') {
             preparedRequest()
-                .get(handler)                    
+                .get(handler)
                 .auth(config.user, config.password)
                 .expect(412, done)
         } else if (verb == 'PUT') {
             preparedRequest()
-                .put(handler)                    
+                .put(handler)
                 .auth(config.user, config.password)
                 .send({})
                 .expect(412, done)
         } else if (verb == 'PATCH') {
             preparedRequest()
-                .patch(handler)                    
+                .patch(handler)
                 .auth(config.user, config.password)
                 .send({})
                 .expect(412, done)
         } else if (verb == 'DELETE') {
             preparedRequest()
-                .delete(handler)                    
+                .delete(handler)
                 .auth(config.user, config.password)
                 .expect(412, done)
         }
-    })     
+    })
 }
-function testAsyncParameter(handler, verb) {
+
+function testAsyncParameter(handler, verb, body) {
     it ('should return 422 if request doesn\'t have the accepts_incomplete parameter', function(done) {
         if (verb == 'PUT') {
             preparedRequest()
                 .put(handler)
                 .set('X-Broker-API-Version', apiVersion)
                 .auth(config.user, config.password)
-                .send({})
+                .send({
+                    service_id: body.service_id || guid.create().value,
+                    plan_id: body.plan_id || guid.create().value,
+                    organization_guid: body.organization_guid || guid.create().value,

```

Comment:
Is this assuming that the request will fail due to the async flag not being there instead of failing due to bogus IDs? I'm not sure you can make the assumption that the broker will always check for the async flag before it checks for valid service/plan IDs.  Basically, I'm not sure you should be generating random IDs instead of using valid values from the broker.

Reply:
Fair point. We can amend the commit and only use the ids provided in the test config json. On the other hand, we are only generating IDs when those IDs are not configured, similar to what you can find [here](https://github.com/openservicebrokerapi/osb-checker/blob/e8be866c73a7fe47140761dbc774a879e5cfb8db/2.13/tests/test/test.js#L60-L62) and [here](https://github.com/openservicebrokerapi/osb-checker/blob/e8be866c73a7fe47140761dbc774a879e5cfb8db/2.13/tests/test/test.js#L335-L340).

Reply:
Do we have tests that verify the broker will reject requests went sent an unknown ids?  I guess its needed for all ops - meaning bad service, plan, instance and binding ids.  Oh, and I guess bad "operation" string too.

Reply:
There are tests for the validity of the service_id and plan_id for the [provision](https://github.com/openservicebrokerapi/osb-checker/blob/e8be866c73a7fe47140761dbc774a879e5cfb8db/2.13/tests/test/test.js#L111-L130), but not for the other cases.

We will amend this PR to send only valid GUIDs and IDs.